core/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -165,6 +168,7 @@ public static void registerPrototype(String type, Custom proto) {
     private final ImmutableOpenMap<String, IndexMetaData> indices;
     private final ImmutableOpenMap<String, IndexTemplateMetaData> templates;
     private final ImmutableOpenMap<String, Custom> customs;
+    private final IndexGraveyard indexGraveyard;
```

Comment:
Can we add a custom type and have `indexGraveyard()` be a syntactic sugar for getting it? we will get a lot of things for free..